src/http/Message.cc

```diff
@@ -95,20 +101,7 @@ Http::Message::parse(const char *buf, const size_t sz, bool eof, Http::StatusCod
         return false;
     }
 
-    if (hdr_len > Config.maxReplyHeaderSize || (hdr_len <= 0 && sz > Config.maxReplyHeaderSize)) {
-        debugs(58, DBG_IMPORTANT, "Too large reply header (" << hdr_len << " > " << Config.maxReplyHeaderSize);
-        *error = Http::scHeaderTooLarge;
-        return false;
-    }
-
-    if (hdr_len <= 0) {
-        debugs(58, 3, "failed to find end of headers (eof: " << eof << ") in '" << buf << "'");
-
-        if (eof) // iff we have seen the end, this is an error
-            *error = Http::scInvalidHeader;
-
-        return false;
-    }
+    assert(hdr_len > 0); // sanityCheckStartLine() rejects unparsed headers
```

Comment:
To be specific sanityCheckStartLine() rejects buffers that do not start with RFC7230 section 3.1 "Start Line" syntax. That includes HTTP/0.9, HTTP/1.1, ICY and some other protocols with almost identical syntax (eg ICAP, SMTP)


```suggestion
    assert(hdr_len > 0); // sanityCheckStartLine() rejects buffers that cannot be parsed
```